src/server/services/collection.service.ts

```diff
@@ -116,3 +120,28 @@ export const upsertCollection = async ({
     },
   });
 };
+
+export const getUserCollectionsByItem = ({
+  input,
+  user,
+}: {
+  input: GetUserCollectionsByItemSchema;
+  user: SessionUser;
+}) => {
+  const { modelId, imageId, articleId, postId } = input;
+
+  return dbRead.collection.findMany({
+    where: {
+      userId: user.id,
```

Comment:
I think this is not enough, consdering you need to think of permissions.

First, you need to:
`getUserCollectionsWithPermissions` (Likely Write/Manage permission in this case, I doubt we care for View)

Then use `where { id: { in: } }` in this query to filter out only those by this items.

Otherwise, you'll only return collections created by the user instead of those he can add/manage on